Detectors/TRD/base/src/TrackletTransformer.cxx

```diff
@@ -75,10 +75,11 @@ float TrackletTransformer::calculateDy(int detector, int slope, const PadPlane*
   return calibratedDy;
 }
 
-float TrackletTransformer::calibrateX(double x) const
+float TrackletTransformer::calibrateX(int detector, double x) const
 {
-  // Hard-coded value will be replaced once t0 calibration is available from CCDB
-  float t0Correction = -0.279;
+  // NOTE: Calibration value is average over all chambers, stored in PHOS hole (chamber 435)
+  //       option to use chamber-wise values will be implemented in the future
+  float t0Correction = mCalT0->getT0(435);
```

Comment:
What is the value for the average T0 of all chambers? Does it spread too much for all chambers or why should we not directly use `detector` here instead? I don't remember now whether you had shown this in the past, sorry

Reply:
The average is 1.223 with a std. dev. of ~0.16 across chambers that don't 1) have a "0" as correction (no PH spectrum or failed fit) or 2) have a bad fit (ie. can be easily identified when looking at the fits). A few chambers deviate a lot farther, mostly due to non-ideal fits because of lacking statistics. With @tdietel we discussed keeping the chamber-wise values for all successfull fits and use the average value for all other chambers; but it was decided to leave 0 for failing fits etc., store the average value in the PHOS hole, and use the average value for all chambers (for now, with the option to maybe later use chamber-wise values)

Reply:
OK, but which unit is that? Probably time bins, right? Because then we would still need to multiply this with the vDrift to get the correction for the radial position.
With the current change you would shift all TRD space points more than 1.5cm towards larger radii compared to what we are doing now.

Reply:
Would it not be elegant to simply add an average value to the CalT0 class, than hide an average value in the PHOS hole.

Reply:
Can the mean be calculated from the other values in the CalT0 class? All the values are there, but are there any criteria that are no longer accessible at this point that are used to include/exclude a value in the calculation of the mean? I am mainly thinking about quality of the fit, i.e. chi2. 
If this is not possible, I would agree with Sean's comment that a separate member variable in CalT0 is a cleaner implementation.

Reply:
@martenole, I do not think that T0 should be multiplied with vDrift, as a change of vDrift does not change the start time of the signal. But we have to make sure T0 corresponds to the correct position within the chamber. We agreed that the T0 calibration will calculate the position of the maximum of the peak. This is necessary, because we do not see the actual start of the signal (unless we can find something in the high-resolution pulse height plots that are now possible with the pretrigger phases). The question is now what this T0 value corresponds to. My guess would be that this is the beginning of the drift region, i.e. just at the time when all the charge from the amplification region has been collected.

Reply:
@luisabergmann what is the unit of your fit results?

Reply:
@martenole The unit is time bins
@tdietel At the moment chi2 of the fits is at this time no longer accessible. We could calculate the average in the TrackletTransformer.cxx if we set the t0 value of bad fits to eg. 0 earlier in the automated process (and as I did manually right now) and then only use the non-zero chambers

Reply:
Of course we could also add members to the `CalT0` class to either store the average directly or to keep chi2 etc to do the selection in `TrackletTransformer.cxx`.

Reply:
Did you check the stability of the fit procedure? This could be done by looking at the correlation of T0 values for all chambers for two calibration time windows. Basically, the question is to which degree the T0 values are limited by the resolution of the calibration algorithm, and to which degree it's differences between the chambers. This also informs if we should use the average or chamber-specific values.